project_setup/scripts/terraform-to-secrets

```diff
@@ -98,6 +98,61 @@ def get_terraform_state(filename: str = None) -> Dict:
     return json_state
 
 
+def find_tagged_secret(resource_name: str, resource_data: Dict) -> Tuple[str, str]:
+    """Extract a tagged secret from a resource."""
+    # Ensure "tags" key exists in resource_data and if it does, make sure
+    # its value is not None.  Both of these cases can occur.
+    tags: Dict[str, str]
+    if "tags" not in resource_data or resource_data.get("tags") is None:
+        tags = dict()
+    else:
+        tags = resource_data["tags"]
+
+    secret_name: Optional[str] = tags.get(GITHUB_SECRET_NAME_TAG)
+    lookup_tag: Optional[str] = tags.get(GITHUB_SECRET_TERRAFORM_LOOKUP_TAG)
+    secret_value: Optional[str]
+    if secret_name:
+        logging.debug(
+            f"Found {GITHUB_SECRET_NAME_TAG} on {resource_name} "
+            f"with value {secret_name}"
+        )
+        if lookup_tag:
+            logging.debug(
+                f"Found {GITHUB_SECRET_TERRAFORM_LOOKUP_TAG} on "
+                f"{resource_name} with value {lookup_tag}"
+            )
+            secret_value = resource_data.get(lookup_tag)
+            if secret_value is None:
+                logging.warning(f"Could not lookup value with key {lookup_tag}")
+            else:
+                logging.debug(f"Looked up value: {secret_value}")
+                return secret_name, secret_value
+        else:
+            logging.warning(
+                f"Missing {GITHUB_SECRET_TERRAFORM_LOOKUP_TAG} on " f"{resource_name}."
+            )
+    return "", ""
```

Comment:
I'm suspicious of this line.   I know its there because of the type checking.  I think we should apply the same design pattern used elsewhere here... That is make the function a generator.  Even though it will only return `1` or `0` tuples, we can then remove the `if secret_name` whenever it is called below, and just use a `for` loop like we do everywhere else.  

As a bonus, if we ever allow resources or users set more than one secret it'll "Just Work™".

Reply:
This is a Good Idea- I will do it.  Stay tuned.

Reply:
@felddy, check out https://github.com/cisagov/development-guide/pull/12/commits/7028f336701c68781a35b6c97ea5421f4a5965ec - it smells much nicer than the previous version.  Thanks for the suggestion!